app/main/routes.py

```diff
@@ -174,20 +180,83 @@ def register():
 
 @main_bp.route('/user_owner', methods=['GET', 'POST'])
 @token_auth.login_required
-def user_owner(token):
+def user_owner():
+    token = request.get_json()['token']
     user = get_one(Owner, 'token', token)
-
     if user:
         return jsonify(user.to_dict())
     return error_response(404)
 
 
 @main_bp.route('/user_sitter', methods=['GET', 'POST'])
 @token_auth.login_required
-def user_sitter(token):
+def user_sitter():
     token = request.get_json()['token']
     user = get_one(Sitter, 'token', token)
     if user:
         return jsonify(user.to_dict())
     return error_response(404)
 
+
+
+
+
+@main_bp.route('/upload_profile_image', methods=['GET', 'POST'])
+@token_auth.login_required
+def upload_file():
+    file = request.files['file']
+    file.filename = secure_filename(file.filename)
+    output   	  = upload_file_to_s3(file, current_app.config["S3_BUCKET"])
+    token = request.headers['Authorization'].split(" ")[1]
+    collection = Sitter if Sitter.check_token(token) else Owner
```

Comment:
this should be happening in the decorator (I think it does)

Reply:
Discussing this as soon as possible